### DISCUSSION TITLE
Altura de um iframe

### DISCUSSION THREAD

**Discussion by reginaldojunior** [frontendbr/forum#1056] [Perguntas e Respostas]
Tem algum jeito menos "gambiarento" de colocar a altura de um iframe automática a partir do conteudo do iframe?

**Comment by brunover** [comment]
Depende do que você quer dizer com *gambiarrento*, hehehehe. Vou dizer como eu fiz, infelizmente não conheço forma mais elegante, se alguém souber uma forma melhor vai ser bom que também aprendo :)

* Adicionei um listener `onload` no _iframe_ que chamava uma função "resizeframe()". Assim que o _iframe_ era carregado a função atualizava a altura dele. Exemplo:

```html
<iframe class="iframe-class" id="iFrameID" src="iframe.html" frameborder="0" scrolling="no" onload="resizeIframe()"></iframe>
```

```javascript
const resizeIframe = () => {
	let iFrameElement = document.getElementById('iFrameID');
	iFrameElement.style.height = iFrameElement.contentWindow.document.body.scrollHeight + "px";
}
```

**Comment by bizarro** [comment]
Eu usava o [iframe-resizer](https://github.com/davidjbradshaw/iframe-resizer), talvez possa ajudar mais ainda caso você use domínios diferentes também. Esqueci de falar que não deixa de ser uma gambiarra, porque ele usa dois scripts. 😅

**Comment by reginaldojunior** [comment]
Não deu totalmente certo, pq o conteudo dos iframes são relativos, tipo eles carregam com um tamanho e dps que ocorre alguma ação do usuario ele carrega uma tabela x e fico de outro tamanho.

**Comment by bizarro** [comment]
Não sei. 🤔 No [exemplo](http://davidjbradshaw.com/iframe-resizer/example/) deles parece funcionar mesmo com alturas diferentes dinâmicas, se você clicar no "Toggle Content".

**Comment by luizbills** [comment]
A altura do iframe é sempre fixa ou fica variando?

Se for fixa você pode usar essa técnica: https://blog.theodo.fr/2018/01/responsive-iframes-css-trick/

Se for váriável, vai ter que ser com javascript. Apenas espere o iframe terminar de carregar e pega a altura do body do contepudo e seta no iframe.

**Comment by kvnol** [comment]
@reginaldojunior, nesse exemplo do @brunover (que adorei e vou usar), talvez não tenha dado certo porque tu não adicionou o listener.

**Comment by marcusvcribeiro** [comment]
Cara, estou com o mesmo problema, se tiver achado alguma solução, me diga.

2018-04-30 14:01 GMT-03:00 Kevin Oliveira <notifications@github.com>:

**Comment by ninetails** [comment]
Tem o [Pym](http://blog.apps.npr.org/pym.js/) que sites, geralmente de mídias/jornais, usam para deixar iframes responsivos. Creio que também redimensiona a altura mas não tenho certeza.
